app/exchanges/parsers/pacbio_sample_sheet_v13_parser.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module Parsers
+  # Parsers::PacbioSampleSheetV13Parser
+  # Use to parse PacBio sample sheets of version 13
+  # Example usage:
+  #    parsed_sample_sheet = Parsers::PacbioSampleSheetV13Parser.new.parse(sample_sheet_string)
+  class PacbioSampleSheetV13Parser
+    # these keys (columns) are excluded because they break value-per-well mapping and
+    # aren't used in our implementation
+    EXCLUDE_KEYS = ['Pipeline Id', 'Analysis Name', 'Entry Points', 'Task Options'].freeze
+
+    def split_into_sections(sample_sheet_string)
+      # returns an hash of sections and their contents as strings
+      # {
+      #   'Run Settings' => 'content',
+      #   'SMRT Cell Settings' => 'content',
+      #   'Samples' => 'content'
+      # }
+      sections = {}
+
+      sample_sheet_string.split(/(?>^|\n)\[/).each do |section|
+        next if section.empty?
+
+        section_name = section.match(/(.*)\]/)[1]
+        section_content = section.gsub(/.*\]/, '').strip
+        sections[section_name] = section_content
+      end
+
+      sections
+    end
+
+    def parse_run_settings(section_content)
+      # returns a hash of key-value pairs
+      # split on newlines, then split on commas
+
+      run_settings = {}
```

Comment:
You can use tap here.

```
{}.tap do |run_setttings|
  section_content.split("\n").each do |line|
    key, value = line.split(',', -1)
    run_settings[key] = value
  end
end
```

tap returns the object so that will be returned from the function.